js/src/stringCollector.js

```diff
@@ -44,32 +46,64 @@ export class StringCollector {
         this.runs.push('\n')
         break
       case BreakType.DOUBLE:
-        this.runs.push('\n\n')
+        let paragraphBreakAdded = false
+        // iterate through runs backwards:
+        for (let i = this.runs.length - 1; i >= 0; i--) {
+          const run = this.runs[i]
+          if (run === '\n\n') {
```

Comment:
Can't we encouter triple+ new line symbols?

Reply:
No, we only insert single and double new lines "manually", and the text only ever has single new lines in it.